src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -2026,12 +2026,10 @@ public static string Match(string text)
                 int longestPossiblePattern = 0;
                 for (int i = 0; i < text.Length; i++)
                 {
+                    // Notice, '-' == ('-' | 0x20)
                     uint h = text[i];
-                    if (h >= 'A' && h <= 'Z')
-                    {
-                        h |= 0x20; // ToLower
-                    }
-                    else if (!((h >= 'a' && h <= 'z') || h == '-'))
+                    h |= 0x20;
```

Comment:
Will blindly doing this cause any problem? `h` could be any character, not just ASCII ones.
The original code is way more readable than this, and I double if the perf gain is worth sacrificing the readability.